parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -211,8 +211,31 @@ loop = do
         eval . Eval $ Branch.getMaybePatch seg (Branch.head b)
       getHQ'TermsIncludingHistorical p =
         getTermsIncludingHistorical (resolveSplit' p) root0
+
+      -- HHQS' -> Term References (for `replace`)
+      getHQ'TermReferences :: Path.HQSplit' -> Set Reference
+      getHQ'TermReferences p =
+        Set.fromList [ r | Referent.Ref r <- toList (getHQ'Terms p) ]
+      -- HHQS' -> Term Referents (for `alias`)
+      getHQ'Terms :: Path.HQSplit' -> Set Referent
       getHQ'Terms p = BranchUtil.getTerm (resolveSplit' p) root0
+      -- HHQS' -> Type References (for `replace` & `alias`)
+      getHQ'Types :: Path.HQSplit' -> Set Reference
       getHQ'Types p = BranchUtil.getType (resolveSplit' p) root0
+      resolveHHQS'Types :: HashOrHQSplit' -> Action' m v (Set Reference)
+      resolveHHQS'Types = either 
+        (eval . TypeReferencesByShortHash)
+        (pure . getHQ'Types)        
+      resolveHHQS'Terms = either
+        (eval . TermReferencesByShortHash)
+        (pure . getHQ'TermReferences)
+      resolveHHQS'Referents = either
+        (\h ->
+          (<>) <$> eval (TermReferentsByShortHash h)
+               <*> fmap (Set.map Referent.Ref)
+                        (eval (TermReferencesByShortHash h))
+               )
+        (pure . getHQ'Terms)
```

Comment:
Just going to assume these are all correct and that they do something useful.

Reply:
`HashOrHQSplit'` is an `Either`, and these just handle both sides of it using existing helpers.

`resolveHHQS'Referents` I think no longer needs to be a `<>` though.